feat: use media queries for `useIsMobile` hook

karrui opened the pull request:
## Problem
<!-- What problem are you trying to solve? What issue does this close? -->
Use media queries instead of the result of `useBreakpointValue` to trigger layout changes.  May break on virtual browsers, we should test before releasing if we can our hands on such devices.



Supercedes #5247
Closes #5244 hopefully

## Solution
<!-- How did you solve the problem? -->

Documentation for [chakra's useMediaQuery()](https://chakra-ui.com/docs/hooks/use-media-query)

**Breaking Changes** 
<!-- Does this PR contain any backward incompatible changes? If so, what are they and should there be special considerations for release? -->
- [x] No - this PR is backwards compatible

karrui committed:
feat: use media queries for useIsMobile hook

timotheeg commented on frontend/src/hooks/useIsMobile.ts:
```diff
@@ -1,11 +1,9 @@
-    xs: true,
-    md: false,
-  })
+  const [isLargerThanMd] = useMediaQuery(`(min-width: ${BREAKPOINT_VALS.md})`)
```
This is pretty neat!

karrui committed:
fix: use useMediaMatch from rooks package instead

easier to handle, seems like chakraUI's implementation requires some additional things to mock in jest
